src/DiffEqGPU.jl

```diff
@@ -238,7 +218,7 @@ function generate_problem(prob::ODEProblem,u0,p,jac_prototype,colorvec)
     _f = let f=prob.f.f
         function (du,u,p,t)
             version = u isa CuArray ? CUDA() : CPU()
-            @launch version gpu_kernel(f,du,u,p,t)
+            wait(gpu_kernel(version)(f,du,u,p,t;ndrange=size(u,2)))
```

Comment:
do we have to wait afterwards? The next call is going to be a broadcast, so I guess it doesn't matter much, but I am curious whether this is required.

Reply:
Probably not. As long as we can propagate the returned event to any dependent kernel invocations, we can synchronize things more efficiently. What's the dependency graph look like for this set of calls?

Reply:
The only other kernels are broadcasts and reductions

Reply:
We need to add at least a device wait, so that we synchronize above and below against other device activity